Cabal/tests/UnitTests/Distribution/Version.hs

```diff
@@ -147,6 +181,35 @@ instance Arbitrary VersionRange where
       orEarlierVersion' v =
         unionVersionRanges (EarlierVersion v) (ThisVersion v)
 
+---------------------
+-- Version properties
+--
+
+prop_VersionId :: [Int] -> Bool
+prop_VersionId lst0 =
+    (versionNumbers . mkVersion) lst == lst
+  where
+    lst = map abs lst0
```

Comment:
Can use `[NonNegative Int]` and `map getNonNegative` rather than `map abs` to be a bit clearer.